cloudinit/config/schema.py

```diff
@@ -410,34 +418,53 @@ def _get_property_doc(schema: dict, prefix="    ") -> str:
     """Return restructured text describing the supported schema properties."""
     new_prefix = prefix + '    '
     properties = []
-    for prop_key, prop_config in schema.get('properties', {}).items():
-        # Define prop_name and description for SCHEMA_PROPERTY_TMPL
-        description = prop_config.get('description', '')
-
-        # Define prop_name and description for SCHEMA_PROPERTY_TMPL
-        properties.append(
-            SCHEMA_PROPERTY_TMPL.format(
-                prefix=prefix,
-                prop_name=prop_key,
-                description=_parse_description(description, prefix),
-                prop_type=_get_property_type(prop_config),
+    property_keys = [
+        schema.get("properties", {}),
+        schema.get("patternProperties", {}),
+    ]
+
+    for props in property_keys:
+        for prop_key, prop_config in props.items():
+            # Define prop_name and description for SCHEMA_PROPERTY_TMPL
+            description = prop_config.get("description", "")
+
+            # Define prop_name and description for SCHEMA_PROPERTY_TMPL
+            label = prop_config.get("label", prop_key)
```

Comment:
Glad to see this change is non-breaking (falls back to prop_key when label isn't defined).